### PR TITLE
Code cleanup

### PR DESCRIPTION
**Pull request by bjoernricks**

### CODE REVIEW

File: gsad/src/gsad_gmp.c [hunk 1]
```diff
@@ -423,8 +414,8 @@ filter_exists (gvm_connection_t *connection, const char *filt_id)
   entity_t entity;
 
   if (filt_id == NULL
-      || strcmp (filt_id, "0")
-      || strcmp (filt_id, "-2"))
+      || str_equal (filt_id, "0")
+      || str_equal (filt_id, "-2"))
```
Comment (timopollmeier) [inline gsad/src/gsad_gmp.c]: I think you forgot to negate the results here.
Also, you could use FILT_ID_NONE and FILT_ID_USER_SETTING here.

File: gsad/src/gsad_http.c [hunk 2]
```diff
@@ -804,22 +806,22 @@ attach_remove_sid (http_response_t *response, const gchar *sid)
 {
   if (sid)
     {
-      if (strcmp (sid, "0"))
+      if (str_equal (sid, "0"))
```
Comment (timopollmeier) [inline gsad/src/gsad_http.c]: I think there's another negation missing here.

File: gsad/src/gsad_gmp.c [hunk 3]
```diff
@@ -2014,12 +1598,12 @@ get_many (gvm_connection_t *connection, const char *type,
           else
             filt_id = FILT_ID_USER_SETTING;
         }
-      else if ((strcmp (filter, "sort=nvt") == 0)
-               && ((strcmp (type, "note") == 0)
-                   || (strcmp (type, "override") == 0)))
+      else if (str_equal (filter, "sort=nvt")
+               && (str_equal (type, "note")
+                   || str_equal (type, "override") == 0))
```
Comment (timopollmeier) [inline gsad/src/gsad_gmp.c]: It looks like you accidentally left the == 0 in here.

File: gsad/src/gsad_gmp.c [hunk 4]
```diff
@@ -2014,12 +1598,12 @@ get_many (gvm_connection_t *connection, const char *type,
...
         filt_id = FILT_ID_USER_SETTING;
-      else if ((strcmp (filter, "apply_overrides=1") == 0)
-               && (strcmp (type, "task") == 0))
+      else if (str_equal (filter, "apply_overrides=1")
+               && str_equal (type, "task") == 0)
```
Comment (timopollmeier) [inline gsad/src/gsad_gmp.c]: Same here

File: gsad/src/gsad_gmp.c (hunk 1 above)
Comment (bjoernricks) [reply gsad/src/gsad_gmp.c]: Thanks for the hint! Didn't know we introduced those defines already. Nice! I'll check the condition.

File: gsad/src/gsad_http.c (hunk 2 above)
Comment (bjoernricks) [reply gsad/src/gsad_http.c]: If have changed the sequence of the clauses. Should be fine here.

File: gsad/src/gsad_gmp.c (hunk 3 above)
Comment (bjoernricks) [reply gsad/src/gsad_gmp.c]: You are right. I'll fix it.

File: gsad/src/gsad_gmp.c (hunk 1 above)
Comment (bjoernricks) [reply gsad/src/gsad_gmp.c]: Regarding the condition. IMHO it was broken before and my change fixed it. Can you take another look?

File: gsad/src/gsad_gmp.c (hunk 4 above)
Comment (bjoernricks) [reply gsad/src/gsad_gmp.c]: All fixed now. Btw. I am not sure if all conditions in this path are correct. The code is confusing to me

File: gsad/src/gsad_gmp.c (hunk 1 above)
Comment (timopollmeier) [reply gsad/src/gsad_gmp.c]: I've checked the context again and it looks like the condition was wrong.

**Pull request merged**